packages/dds/merge-tree/src/partialLengths.ts

```diff
@@ -931,19 +942,21 @@ export class PartialSequenceLengths {
 				const moveIsLocal = !!moveInfo && moveInfo.movedSeq === UnassignedSequenceNumber;
 
 				const removeHappenedFirst =
-					removalInfo &&
+					removalInfo?.removedSeq !== undefined &&
```

Comment:
Similar to Abram's comment here might prefer revert the conditional change and non-null assert to minimize runtime change.